crates/ruff/src/rules/flake8_annotations/helpers.rs

```diff
@@ -52,19 +28,20 @@ pub(crate) fn is_overload_impl(
     overloaded_name: &str,
     semantic: &SemanticModel,
 ) -> bool {
-    if let Definition::Member(Member {
-        kind: MemberKind::Function | MemberKind::NestedFunction | MemberKind::Method,
-        stmt,
+    let Definition::Member(Member {
+        kind:
+            MemberKind::Function(function)
+            | MemberKind::NestedFunction(function)
+            | MemberKind::Method(function),
```

Comment:
Nit: We have this pattern in multiple places. How about a `definition.is_function_or_method` helper function?